### PR TITLE
ohad aizik HW commit

### PR DESCRIPTION
**Pull request by oaizik**

### CODE REVIEW

File: hw1_interview_questions/ohad_aizik.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      if (tempSequence > maxSequence) {
+        maxSequence = tempSequence;
+      }
+    } else {
+      tempSequence = 0;
+    }
+  }
+
+  return maxSequence;
+}
```
Comment (TomerMizrachi) [inline hw1_interview_questions/ohad_aizik.ts:47]: the code is nice and clear, comments are in place.

File: hw1_interview_questions/__tests__/ohad_aizik-test.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  expect(longestConsecutive([2, 2, 2, 2])).toEqual(1);
+});
+
+test('longestConsecutive of 11 elements array', () => {
+  expect(longestConsecutive([1, 5, 40, 41, 2, 3, 42, 44, 4, 43, 39])).toEqual(6);
+});
+
+test('longestConsecutive of 11 elements array with Duplicates', () => {
+  expect(longestConsecutive([1, 3, 6, 2, 4, 4, 9, 9, 7, 7, 1])).toEqual(4);
+});
```
Comment (TomerMizrachi) [inline hw1_interview_questions/__tests__/ohad_aizik-test.ts:21]: tests file is satisfactory and covers edge cases

**Review by TomerMizrachi — COMMENTED** [review]
review for HW1 Tomer to Ohad

File: hw1_interview_questions/ohad_aizik.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+Constraints:
+0 <= nums.length <= 104
+0 <= nums[i] <= 231 - 1
+*/
+
+export function longestConsecutive(nums: number[]): number {
+  // finding the max element in the array, if the array is empty we use 0
+  const maxNumber: number = nums.length > 0 ? Math.max(...nums) : 0;
+  // creating array with in the max number length initialized with undefined values for the counting sort
+  const countingArray: number[] = new Array(maxNumber).fill(undefined);
```
Comment (yoav-zibin) [inline hw1_interview_questions/ohad_aizik.ts:25]: no need to do .fill(undefined)
(that ruins the O(n) complexity)
